specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2023-12-01-preview/AdvancedThreatProtectionSettings.json

```diff
@@ -0,0 +1,404 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the server's Advanced Threat Protection state.",
+            "schema": {
+              "$ref": "#/definitions/AdvancedThreatProtection"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Location": {
+                "description": "URL to retrieve the final result after operation completes.",
+                "type": "string"
+              },
+              "Azure-AsyncOperation": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Enable a server's Advanced Threat Protection settings": {
+            "$ref": "./examples/AdvancedThreatProtectionSettingsPatchEnabled.json"
+          },
+          "Disable a server's Advanced Threat Protection settings with all parameters": {
+            "$ref": "./examples/AdvancedThreatProtectionSettingsPatchDisabled.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "AdvancedThreatProtectionSettings"
+        ],
+        "description": "Updates a server's Advanced Threat Protection state.",
+        "operationId": "AdvancedThreatProtectionSettings_UpdatePut",
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common-types/v1/common-types.json#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AdvancedThreatProtectionNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The server's Advanced Threat Protection body to update.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AdvancedThreatProtection"
+            }
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

I see that async PUT with both 201 and 202 response codes was approved in the private repo, but I don't understand what it means. Your service can't return both status codes, so which is it actually returning for async PUT (and what is the rationale for putting both in the spec)?

Reply:
I have discussed with Roopesh and get his approval for suppression, since we use a unified framework to construct response. And Roopesh said we can use current pattern until ARM team has a proposal for such issue. And I also post our discussion in teams.

Reply:
That was approval to continue using 202, which is fine. The problem here is declaring both 201 + provisioningState and 202 + Azure-AsyncOperation. Your service can't do both on the same PUT method.

Since your current stable API ([https://github.com/Azure/azure-rest-api-specs/blob/e79c2debdebd659662c78885c117603d991d3b06/specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/stable/2023-06-30/FlexibleServers.json#L81](https://github.com/Azure/azure-rest-api-specs/blob/e79c2debdebd659662c78885c117603d991d3b06/specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/stable/2023-06-30/FlexibleServers.json#L81)) already does this, I'm not blocking approval.

But your RP should fix this in the next API version.